clusterman/kubernetes/kubernetes_cluster_connector.py

```diff
@@ -219,33 +219,37 @@ def drain_node(self, node_name: str, disable_eviction: bool) -> bool:
             return False
 
     def cordon_node(self, node_name: str) -> bool:
+        now = str(arrow.now().timestamp)
         try:
-            self._core_api.patch_node(
-                name=node_name,
-                body={
-                    "spec": {
-                        "unschedulable": True,
-                    },
-                },
+            node = next(filter(lambda x: x.metadata.name == node_name, self._nodes_by_ip.values()))
+            if not node:
+                return False
+            taints = (
+                list(filter(lambda x: x.key != CLUSTERMAN_TERMINATION_TAINT_KEY, node.spec.taints))
+                if node.spec.taints
+                else []
             )
+            taints.append({"effect": "NoSchedule", "key": CLUSTERMAN_TERMINATION_TAINT_KEY, "value": now})
+            self._core_api.patch_node(name=node_name, body={"spec": {"taints": taints}})
             return True
         except ApiException as e:
-            logger.warning(f"Failed to cordon {node_name}: {e}")
+            logger.warning(f"Failed to cordon {node_name}: {e.status} - {e.reason}")
             return False
 
     def uncordon_node(self, node_name: str) -> bool:
         try:
-            self._core_api.patch_node(
-                name=node_name,
-                body={
-                    "spec": {
-                        "unschedulable": False,
-                    },
-                },
+            node = next(filter(lambda x: x.metadata.name == node_name, self._nodes_by_ip.values()))
+            if not node:
+                return False
```

Comment:
On top of the comment above, this should go into a `get_node_by_hostname` method to avoid duplicating code. Or even move the whole thing into a `get_node_taints` method.

Reply:
get_node_by_hostname - done, I will keep taints part duplicate for now. I'll probably merge cordon/uncordon method.